### PR TITLE
CI/CD/Security/Minor change

### PR DESCRIPTION
**Pull request by leviathanch**
This patch introduces a version upgrade of a vulnerable dependency, as well as fixes the CI/CD build for Windows and Ubuntu. Additionally, two unused imports are being removed.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      - name: Linux Cargo Cache
+        id: linux-cache-cargo
+        uses: actions/cache@v4
+        with:
+          path: /home/runner/.cargo
+          key: linux-cache-cargo
+      - name: Install botan
+        run: sudo apt install -y libbotan-2-dev
+      - uses: actions/checkout@v4
+      - run: sudo apt install cargo
```
Comment (codingpeanut15) [inline .github/workflows/release.yml:21]: have a look at this action [here](https://github.com/plancksecurity/planck-core/blob/45fefece7d253342c88c89ec0b301dc6db8a9d07/.github/workflows/check-code-correctness.yml#L21)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      PREFIX: /home/runner/install
+    steps:
+      - name: Linux Cargo Cache
+        id: linux-cache-cargo
+        uses: actions/cache@v4
+        with:
+          path: /home/runner/.cargo
+          key: linux-cache-cargo
+      - name: Install botan
+        run: sudo apt install -y libbotan-2-dev
```
Comment (codingpeanut15) [inline .github/workflows/release.yml:19]: are we ok with installing OEM botan for ubuntu?

**Review by codingpeanut15 — COMMENTED** [review]
- I am not 100% sure that by manually updating Cargo.lock you are actually fixing the dependencies long term. I mean someone can act on the cargo.toml and your change to the cargo lock will get lost. A better option would be to do as the main Cargo.toml does [here](https://github.com/plancksecurity/foundation-planckCoreSequoiaBackend/blob/1e0885fe812356d03d93cbd8e4d3dcd031c78b40/Cargo.toml#L36)
- have you tested the windows build without downloading the botan artifact to see if the build fails? otherwise there might be a pre-installed botan that comes with the windows image that github action uses
- do we need the ubuntu build at all?

File: .github/workflows/release.yml (hunk 2 above)
Comment (ghost) [reply .github/workflows/release.yml:19]: At the moment, the changes we did to handling RNG/seeding are only relevant to windows, but I assume we have to be ready to fine tune this on any other platform as well. So our own version of Botan all the time.

**Comment by leviathanch** [comment]
I've submitted a new pull request which addresses this and makes some more adjustments

**Pull request closed without merging**